src/main/java/io/jenkins/plugins/sample/OnboardingSectionConfiguration.java

```diff
@@ -0,0 +1,245 @@
+package io.jenkins.plugins.sample;
+
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.domains.DomainRequirement;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.Job;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import hudson.util.Secret;
+import jenkins.model.GlobalConfiguration;
+import jenkins.model.Jenkins;
+import org.jenkinsci.plugins.plaincredentials.StringCredentials;
+import org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.verb.POST;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.*;
+import java.util.regex.Pattern;
+
+@Extension
+public class OnboardingSectionConfiguration extends GlobalConfiguration {
+
+    private String name;
+
+    private String description;
+
+    private String url;
+
+    private String username;
+
+    private String password;
```

Comment:
Never use a `String` for secret values that get persisted. Either use the `Secret` type, or typically better is to have the user configure credentials separately and then just have the user configure the ID of the relevant credentials. See https://www.jenkins.io/doc/developer/security/secrets/.

Reply:
yeah, makes sense. I'll update it.